pkg/apis/v1alpha5/machine_disruption_gate_defaults.go

```diff
@@ -0,0 +1,33 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha5
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetDefaults for the MachineDisruptionGate
+func (m *MachineDisruptionGate) SetDefaults(ctx context.Context) {
```

Comment:
Thoughts on skipping validation and defaults? I'm definitely skeptical of this default.

Reply:
I think Validation makes sense, but this default I'm 50/50 on. We definitely want to only affect Karpenter nodes, so we can either rely on injecting this into each list call or default it here. If someone wants to do only specific nodes to a provisioner, injecting it into the list call may have the proper AND semantic, whereas this doesn't enforce that.

Reply:
Convinced myself to inject it into the list call 👍